### PR TITLE
Update mergesort.py

### PR DESCRIPTION
**Pull request by LuisEnMarroquin**
Allow the function to be executed without having to pass the start and end indexes of the array

### **Describe your change:**

Added default values to `mergesort` function to automatically calculate the start and end indexes

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?
* [x] Improved an algorithm?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: divide_and_conquer/mergesort.py [hunk 1]
```diff
@@ -27,13 +27,17 @@ def merge(arr, left, mid, right):
...
+def mergesort(arr, left=None, right=None):
     """
-    >>> mergesort([3, 2, 1], 0, 2)
+    >>> mergesort([3, 2, 1])
     [1, 2, 3]
-    >>> mergesort([3, 2, 1, 0, 1, 2, 3, 5, 4], 0, 8)
+    >>> mergesort([3, 2, 1, 0, 1, 2, 3, 5, 4])
     [0, 1, 1, 2, 2, 3, 3, 4, 5]
     """
+    if left is None:
```
Comment (cclauss) [inline divide_and_conquer/mergesort.py:37]:
```suggestion
    left = left or 0
```
All other falsy values are as equally dangerous as None.

Comment (LuisEnMarroquin) [reply divide_and_conquer/mergesort.py:37]: I agree, GitHub is not allowing me to send an extra commit to this pull request, I'll close this in favor of #2444

**Comment by LuisEnMarroquin** [comment]
Closed in favour of #2444

File: divide_and_conquer/mergesort.py (hunk 1 above)
Comment (cclauss) [reply divide_and_conquer/mergesort.py:37]: In the upper right of the file, you should see three dots (...).  Click that and select `edit`.

Comment (LuisEnMarroquin) [reply divide_and_conquer/mergesort.py:37]: I can't, apparently github doesn't allow adding more commits if you remove your folk

image

**Pull request closed without merging**